blaze/compute/sql.py

```diff
@@ -1208,9 +1209,28 @@ def compute_up(expr, data, **kwargs):
 
 @dispatch(StrCat, ColumnElement, ColumnElement)
 def compute_up(expr, lhs_data, rhs_data, **kwargs):
-    res = sa.sql.functions.concat(lhs_data,
-                                  expr.sep,
-                                  rhs_data).label(expr.lhs._name)
+    """
+    concat two string columns element wise. If a row in either column is NULL,
+    then return a NULL otherwise perform concat() with the user defined 'sep'
+    between the two elements.
+    """
+    if expr.sep is None:
+        # this will automatically handle null values correctly
+        res = (lhs_data + rhs_data).label(expr.lhs._name)
```

Comment:
Use the `+` to concatenate if no sep - it's easier to read. I assume it is faster since SQL doesn't have to process the case statement but mostly added this so we can clearly see what the simple case should be doing.